src/sys/timer.cxx

```diff
@@ -1,82 +1,72 @@
-#include <mpi.h>
 #include <bout/sys/timer.hxx>
+#include <mpi.h>
```

Comment:
Can we use `boutcomm` here instead (promises a header guard)? It might be nice for `boutcomm` to provide a `time` method rather than calling `MPI_Wtime` directly here as this would allow us to easily switch to a non-mpi based timer should we need to (e.g. if we wanted to support a serial build).

Reply:
I would not be opposed to just replacing `MPI_Wtime` with either `std::chrono::high_resolution` or `std::chrono::steady_clock`. I don't think there's any advantage to the MPI function over the C++ ones

Reply:
I seem to recall previously seeing guidance that `MPI_Wtime` was generally the best choice timer where available but I can't remember why so this may have changed.

Reply:
(also http://www.cplusplus.com/forum/general/187899/#msg913531 might be handy -- we're measuring intervals so probably want a steady timer)

Reply:
Thanks! I wasn't sure which to prefer, but that recipe is very nice.

Any objections to replacing `MPI_Wtime` with a C++ steady timer?

Reply:
No not really; it's not critical to the code working correctly, just a useful performance hint.